Sources/NIOSOCKS/Messages/ClientGreeting.swift

```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+/// The SOCKS handshake begins with the client sending a greeting
+/// containing supported authentication methods.
+public struct ClientGreeting: Hashable {
+    
+    /// The SOCKS protocol version - we currently only support v5.
+    public var version: UInt8
+    
+    /// The client's supported authentication methods, defined in RFC 1928.
+    public var methods: [AuthenticationMethod]
+    
+    /// Creates a new client greeting with the given authentication methods.
+    /// - parameter methods: The client's supported authentication methods.
+    public init(methods: [AuthenticationMethod]) {
+        self.version = 5
+        self.methods = methods
+    }
+    
+    init?(buffer: inout ByteBuffer) {
+        guard
+            let version = buffer.readInteger(as: UInt8.self),
```

Comment:
Should we police that this is version 5?

Reply:
We should